response/response.go

```diff
@@ -97,39 +102,108 @@ func SetDesiredComposedResources(rsp *v1beta1.RunFunctionResponse, dcds map[reso
-func Fatal(rsp *v1beta1.RunFunctionResponse, err error) {
+// A corresponding event will be created for the Composite Resource.
+func Fatal(rsp *v1beta1.RunFunctionResponse, err error) *Result {
 	if rsp.GetResults() == nil {
 		rsp.Results = make([]*v1beta1.Result, 0, 1)
 	}
-	rsp.Results = append(rsp.GetResults(), &v1beta1.Result{
+	result := &v1beta1.Result{
 		Severity: v1beta1.Severity_SEVERITY_FATAL,
 		Message:  err.Error(),
-	})
+		Target:   v1beta1.Target_TARGET_COMPOSITE,
+	}
+	rsp.Results = append(rsp.GetResults(), result)
+
+	return &Result{
+		result: result,
+	}
 }
 
 // Warning adds a warning result to the supplied RunFunctionResponse.
-func Warning(rsp *v1beta1.RunFunctionResponse, err error) {
+// A corresponding event will be created for the Composite Resource.
+func Warning(rsp *v1beta1.RunFunctionResponse, err error) *Result {
 	if rsp.GetResults() == nil {
 		rsp.Results = make([]*v1beta1.Result, 0, 1)
 	}
-	rsp.Results = append(rsp.GetResults(), &v1beta1.Result{
+	result := &v1beta1.Result{
 		Severity: v1beta1.Severity_SEVERITY_WARNING,
 		Message:  err.Error(),
-	})
+		Target:   v1beta1.Target_TARGET_COMPOSITE,
+	}
+	rsp.Results = append(rsp.GetResults(), result)
+
+	return &Result{
+		result: result,
+	}
 }
 
 // Normal adds a normal result to the supplied RunFunctionResponse.
-func Normal(rsp *v1beta1.RunFunctionResponse, message string) {
+// A corresponding event will be created for the Composite Resource.
+func Normal(rsp *v1beta1.RunFunctionResponse, message string) *Result {
 	if rsp.GetResults() == nil {
 		rsp.Results = make([]*v1beta1.Result, 0, 1)
 	}
-	rsp.Results = append(rsp.GetResults(), &v1beta1.Result{
+	result := &v1beta1.Result{
 		Severity: v1beta1.Severity_SEVERITY_NORMAL,
 		Message:  message,
-	})
+		Target:   v1beta1.Target_TARGET_COMPOSITE,
+	}
+	rsp.Results = append(rsp.GetResults(), result)
+
+	return &Result{
+		result: result,
+	}
 }
 
 // Normalf adds a normal result to the supplied RunFunctionResponse.
-func Normalf(rsp *v1beta1.RunFunctionResponse, format string, a ...any) {
-	Normal(rsp, fmt.Sprintf(format, a...))
+// A corresponding event will be created for the Composite Resource.
+func Normalf(rsp *v1beta1.RunFunctionResponse, format string, a ...any) *Result {
+	return Normal(rsp, fmt.Sprintf(format, a...))
+}
+
+// TargetComposite configures the Composite to receive any events or conditions
+// generated by this result.
+func (o *Result) TargetComposite() *Result {
+	o.result.Target = v1beta1.Target_TARGET_COMPOSITE
+	return o
+}
+
+// TargetCompositeAndClaim configures both the Claim and Composite to receive
+// any events or conditions generated by this result.
+func (o *Result) TargetCompositeAndClaim() *Result {
+	o.result.Target = v1beta1.Target_TARGET_COMPOSITE_AND_CLAIM
+	return o
+}
+
+// ConditionTrue configures the result to create a condition on the targeted
+// objects with the status set to true.
+func (o *Result) ConditionTrue(t string, r string) *Result {
```

Comment:
```suggestion
func (o *Result) ConditionTrue(typ, reason string) *Result {
```

I'm usually all for short variables when the type provides a hint as to what they are (e.g. `t xpv1.ConditionType`). Here where they're both `string` I think longer names are warranted.

Unfortunately we can't use `type` since it's a reserved word in Go. We could use `xpv1.ConditionType` (a subtype of `string`) if we wanted. The downside is folks would need to type convert every time they called this method.